src/backend/distributed/executor/executor.c

```diff
@@ -1547,27 +1547,30 @@ ManageWorkerPool(WorkerPool *workerPool)
 	}
 	else
 	{
-		if (initiatedConnectionCount >= targetPoolSize)
-		{
-			/* already reached the minimal pool size */
-			return;
-		}
+		/* cannot open more than targetPoolSize connections */
+		int maxNewConnectionCount = targetPoolSize - initiatedConnectionCount;
```

Comment:
Maybe `maxNewConnectionCount ` to `maxAllowedConnectionCount `

Reply:
I actually changed it from `maxAllowedConnectionCount` to `maxNewConnectionCount`  :)

I figured the maximum allowed connection count is the pool size, while here we're calculated the maximum number of new connections that we're allowed to open according to the pool size.